src/Microsoft.AspNetCore.Http.Features/IHttpResponseTrailersFeature.cs

```diff
@@ -0,0 +1,30 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Http.Features
+{
+    public interface IHttpResponseTrailersFeature
+    {
+        /// <summary>
+        /// The response trailers to send.
+        /// </summary>
+        IHeaderDictionary Trailers { get; set; }
+
+        /// <summary>
+        /// Indicates if the server has started writing the response trailers.
+        /// If true, the <see cref="Trailers"/> are now immutable, and <see cref="OnStarting"/> should no longer be called.
+        /// </summary>
+        bool HasStarted { get; }
```

Comment:
WasWritten?

Reply:
That would indicate that it's set _after_ the trailers has been written? This is set just before the server starts writing.

Reply:
I'm not too convinced WasWritten is a better name either.

I proposed WasWritten since the Trailers dictionary is closed for modification and the server has committed to writing what's in the dictionary as trailers at the point this property is set. From the perspective of the app developer, I don't see how committing to writing the trailers is observably different from actually writing the trailers especially given that there's write-behind response buffer.

HasStarted isn't bad though. I was thinking HasStartedWriting to make it less ambiguous, but I suggested WasWritten since that's shorter.

If we decided to change OnStarting to OnStartingTrailers or something similar, we might want to change this to HasStartedTrailers instead.

Reply:
Agan, I chose `HasStarted` because it's consistent with `IHttpResponseFeature.HasStarted`. If we can find another name, or get rid of the property entirely, that's cool with me.

Reply:
IsReadOnly? IHeaderDictionary already has that property, we can probably remove this one.

Reply:
Why would the trailer dictionary not be mutable? I think I'm missing something in the IsReadOnly comment